orchestra/actions/util.py

```diff
@@ -38,24 +46,143 @@ def run_script(script,
 
     script_to_run += script
 
-    if quiet:
+    if capture_output:
         stdout = subprocess.PIPE
         stderr = subprocess.PIPE
     else:
-        logger.info(f"The following script is going to be executed:\n" + script.strip())
-        logger.info(f"Script output:")
         stdout = None
         stderr = None
 
+    logger.log(loglevel, f"The following script is going to be executed:\n" + script.strip())
     result = subprocess.run(["/bin/bash", "-c", script_to_run], stdout=stdout, stderr=stderr, cwd=cwd)
+
+    if capture_output:
+        if stdout_loglevel:
+            logger.log(stdout_loglevel, f"STDOUT:")
+            logger.log(stdout_loglevel, try_decode(result.stdout))
+        if stderr_loglevel:
+            logger.log(stdout_loglevel, f"STDERR:")
+            logger.log(stderr_loglevel, try_decode(result.stderr))
+
     if check_returncode and result.returncode != 0:
         raise OrchestraException(f"Script failed with return code {result.returncode}")
 
     return result
 
 
+def run_internal_script(script, environment: OrderedDict = None):
+    """Helper for running internal scripts.
+    :param script: the script to run
+    :param environment: optional additional environment variables
+    :return: a subprocess.CompletedProcess instance
+    """
+    return run_script(
+        script,
+        environment=environment,
+        loglevel="DEBUG",
+        capture_output=True,
+        stdout_loglevel="DEBUG",
+        stderr_loglevel="DEBUG",
+    )
+
+
+def run_user_script(script, environment: OrderedDict = None):
+    """Helper for running user scripts
+    :param script: the script to run
+    :param environment: optional additional environment variables
+    :return: a subprocess.CompletedProcess instance
+    """
+    return run_script(
+        script,
+        environment=environment,
+        loglevel="INFO",
+        capture_output=False,
+    )
+
+
+def get_script_output(script, environment: OrderedDict = None, check_returncode=True, decode_as="utf-8"):
+    """Helper for getting stdout of a script
+    :param script: the script to run
+    :param environment: optional additional environment variables
+    :param check_returncode: if True, raise an exception if the script returns a nonzero exit code
+    :param decode_as: decode the script output using this encoding
+    :return: the stdout produced by the script
+    """
+    return run_script(
+        script,
+        environment=environment,
+        loglevel="DEBUG",
+        capture_output=True,
+        check_returncode=check_returncode,
+    ).stdout.decode(decode_as)
+
+
+def run_subprocess(
```

Comment:
I think this should be a `_run_subprocess` to stress that you're not supposed to call it directly.